huaweicloud/services/dbss/resource_huaweicloud_dbss_instance.go

```diff
@@ -558,6 +555,37 @@ func resourceInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta in
 		instanceId = d.Id()
 	)
 
+	client, err := cfg.NewServiceClient("dbss", region)
+	if err != nil {
+		return diag.Errorf("error creating DBSS client: %s", err)
+	}
+
+	if d.HasChanges("name", "description") {
+		updateHttpUrl := "v1/{project_id}/dbss/audit/instances/{instance_id}"
+		updatePath := client.Endpoint + updateHttpUrl
+		updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+		updatePath = strings.ReplaceAll(updatePath, "{instance_id}", d.Get("instance_id").(string))
+
+		updateOpts := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+		}
+		updateOpts.JSONBody = buildUpdateBodyParams(d)
+		resp, err := client.Request("PUT", updatePath, &updateOpts)
+		if err != nil {
+			return diag.Errorf("error updating DBSS instance: %s", err)
+		}
+
+		respBody, err := utils.FlattenResponse(resp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		result := utils.PathSearch("result", respBody, "").(string)
+		if result != "success" {
+			return diag.Errorf("failed to update the DBSS instance")
+		}
```

Comment:
The logic for updating the name and description fields can be extracted separately as a method, similar to updating the security group ID